tests/Test/AbstractFixerTestCase.php

```diff
@@ -54,36 +46,22 @@ protected function setUp()
         $this->fixer = $this->createFixer();
     }
 
-    /**
-     * @return FixerInterface
-     */
-    protected function createFixer()
+    protected function tearDown()
     {
-        $fixerClassName = $this->getFixerClassName();
-
-        return new $fixerClassName();
-    }
+        parent::tearDown();
 
-    /**
-     * Create fixer factory with all needed fixers registered.
-     *
-     * @return FixerFactory
-     */
-    protected function createFixerFactory()
-    {
-        return FixerFactory::create()->registerBuiltInFixers();
+        $this->linter = null;
+        $this->fixer = null;
     }
 
     /**
-     * @return string
+     * @return FixerInterface
      */
-    protected function getFixerName()
+    protected function createFixer()
     {
-        $reflection = new \ReflectionClass($this);
+        $fixerClassName = preg_replace('/^(PhpCsFixer)\\\\Tests(\\\\.+)Test$/', '$1$2', get_called_class());
 
-        $name = preg_replace('/FixerTest$/', '', $reflection->getShortName());
-
-        return Utils::camelCaseToUnderscore($name);
+        return new $fixerClassName();
```

Comment:
well, it works only because we follow convention super strictly, if one would create `FooFixer` that `getName` would return `bar_baz`, this implementation wouldn't work, while original one would remain to work

just pointing this out (if we ever hit that problem).

i'm fine with the change

Reply:
Maybe better to add a (final?) `testGetName()` method in this abstract class?